include/envoy/upstream/upstream.h

```diff
@@ -38,14 +38,24 @@ class Host : virtual public HostDescription {
     HostDescriptionConstSharedPtr host_description_;
   };
 
-  enum class HealthFlag {
-    // The host is currently failing active health checks.
-    FAILED_ACTIVE_HC = 0x1,
-    // The host is currently considered an outlier and has been ejected.
-    FAILED_OUTLIER_CHECK = 0x02,
-    // The host is currently marked as unhealthy by EDS.
-    FAILED_EDS_HEALTH = 0x04,
-  };
+  // We use an X-macro here to make it easier to verify that all the enum values are accounted for.
+// clang-format off
+#define HEALTH_FLAG_ENUM_VALUES(m)                                               \
+  /* The host is currently failing active health checks. */                      \
+  m(FAILED_ACTIVE_HC, 0x1)                                                       \
+  /* The host is currently considered an outlier and has been ejected. */        \
+  m(FAILED_OUTLIER_CHECK, 0x02)                                                  \
+  /* The host is currently marked as unhealthy by EDS. */                        \
+  m(FAILED_EDS_HEALTH, 0x04)                                                     \
+  /* The host is currently marked as degraded through active health checking. */ \
+  m(DEGRADED_ACTIVE_HC, 0x08)
+  // clang-format on
```

Comment:
nit: outdent

Reply:
Turns out clang-format won't let me outdent it, so I indented `clang-format on` to match this one instead.

Reply:
hmm ok that's strange. No matter.

Reply:
Sorry, actually, why disable clang-format at all here out of curiosity?

Reply:
This is what it looks like when clang formatted:

```
#define HEALTH_FLAG_ENUM_VALUES(m)                                                                 \
  /* The host is currently failing active health checks. */                                        \
  m(FAILED_ACTIVE_HC, 0x1) /* The host is currently considered an outlier and has been ejected. */ \
      m(FAILED_OUTLIER_CHECK, 0x02) /* The host is currently marked as unhealthy by EDS. */        \
      m(FAILED_EDS_HEALTH,                                                                         \
        0x04) /* The host is currently marked as degraded through active health checking. */       \
      m(DEGRADED_ACTIVE_HC, 0x08)
```